bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/scheduling/Scheduler.java

```diff
@@ -462,6 +480,13 @@ public FutureValue createFuture(Strand parent, Callback callback, Map<String, Ob
         return createFuture(parent, callback, constraint, newStrand);
     }
 
+    public FutureValue createTransactionalFuture(Strand parent, Callback callback, Map<String, Object> properties,
+                                    Type constraint, String name, StrandMetadata metadata) {
+        Strand newStrand = new Strand(name, metadata, this, parent, properties, parent != null ?
+                parent.currentTrxContext : null);
```

Comment:
Can parent can be null since transaction function can be call from either main strand or resource strand.

Reply:
Parent can be null, hence the above if check